### PR TITLE
change: Add json schema validation for dwave device schemas.

### PR DESCRIPTION
**Pull request by math411**
*Issue #, if available:*

*Description of changes:*
change: Add json schema validation for dwave device schemas.
*Testing done:*
Updated unit tests
## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ x ] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [ x ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [ x ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ x ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ x ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/unit_tests/braket/aws/test_aws_quantum_task.py [hunk 1]
```diff
@@ -414,55 +415,235 @@ def test_from_circuit_with_shots_value_error(aws_session, arn, circuit):
...
+                    }
+                }
             ),
             "arn:aws:braket:::device/qpu/d-wave/DW_2000Q_6",
         ),
-        (
+        pytest.param(
             {},
             "arn:aws:braket:::device/qpu/d-wave/DW_2000Q_6",
+            marks=pytest.mark.xfail(reason="Missing device params"),
```
Comment (ajberdy) [inline test/unit_tests/braket/aws/test_aws_quantum_task.py:646]: Are you sure this should fail? IIRC, all of the device parameters are optional, so an empty dict should succeed. I suspect this is only passing as a unit test because xfail doesn't fail the testing suite unless `strict=True`.

File: test/unit_tests/braket/aws/test_aws_quantum_task.py [hunk 2]
```diff
@@ -480,14 +661,16 @@ def test_from_annealing(device_parameters, aws_session, arn, problem):
     assert task == AwsQuantumTask(
         mocked_task_arn, aws_session, AnnealingQuantumTaskResult.from_string
     )
-
+    annealing_parameters = _create_annealing_device_params(device_parameters, device_arn=arn)
+    # validate(json.loads(device_parameters.json()), annealing_parameters.schema())
```
Comment (ajberdy) [inline test/unit_tests/braket/aws/test_aws_quantum_task.py:665]: You can delete this comment

File: test/unit_tests/braket/aws/test_aws_quantum_task.py [hunk 3]
```diff
@@ -414,55 +415,235 @@ def test_from_circuit_with_shots_value_error(aws_session, arn, circuit):
...
+                    "initialState": [1, 3, 0, 1],
+                    "maxResults": 1,
+                    "programmingThermalizationDuration": 625,
+                    "readoutThermalizationDuration": 256,
+                    "reduceIntersampleCorrelation": False,
+                    "reinitializeState": True,
+                    "resultFormat": "RAW",
+                    "spinReversalTransformCount": 100,
+                }
+            },
```
Comment (ajberdy) [inline test/unit_tests/braket/aws/test_aws_quantum_task.py:438]: I like that you expanded these mocks to be more realistic

File: test/unit_tests/braket/aws/test_aws_quantum_task.py [hunk 4]
```diff
@@ -480,14 +661,16 @@ def test_from_annealing(device_parameters, aws_session, arn, problem):
     assert task == AwsQuantumTask(
         mocked_task_arn, aws_session, AnnealingQuantumTaskResult.from_string
     )
-
+    annealing_parameters = _create_annealing_device_params(device_parameters, device_arn=arn)
+    # validate(json.loads(device_parameters.json()), annealing_parameters.schema())
+    validate(json.loads(annealing_parameters.json()), annealing_parameters.schema())
```
Comment (ajberdy) [inline test/unit_tests/braket/aws/test_aws_quantum_task.py:666]: The validation should occur against the schema class, not against the annealing parameters. E.g. `Dwave2000QDeviceParameters.schema()`. It is probably easiest to add this as another variable passed in with each parametrization. 

Also, to mirror how the parameters are passed to the api, the json call should have the keyword `exclude_none=True`.

Comment (math411) [reply test/unit_tests/braket/aws/test_aws_quantum_task.py:666]: Added in the exclude. The schema change was resulting in failures

File: test/unit_tests/braket/aws/test_aws_quantum_task.py [hunk 5]
```diff
@@ -414,53 +415,232 @@ def test_from_circuit_with_shots_value_error(aws_session, arn, circuit):
...
+                        "readoutThermalizationDuration": 256,
+                        "reduceIntersampleCorrelation": False,
+                        "reinitializeState": True,
+                        "resultFormat": "RAW",
+                        "spinReversalTransformCount": 100,
+                    }
+                }
             ),
             "arn:aws:braket:::device/qpu/d-wave/DW_2000Q_6",
         ),
```
Comment (ajberdy) [inline test/unit_tests/braket/aws/test_aws_quantum_task.py:642]:
```suggestion
        ),
        (
            Dwave2000QDeviceParameters.parse_obj(
                {"deviceLevelParameters": {}}
            ),
            "arn:aws:braket:::device/qpu/d-wave/DW_2000Q_6",
        ),
```

Wouldn't hurt to add a test for empty/incomplete device level parameters

**Review by ajberdy — APPROVED** [review]
